app/controllers/admin/volunteer_events_controller.rb

```diff
@@ -35,11 +35,13 @@ def invite
     @volunteer_event = @event.volunteer_events.find_or_initialize_by(volunteer: @volunteer)
     new_record = @volunteer_event.new_record?
     @volunteer_event.invite!
-    NewEventMailer.invite_volunteer(@event, @volunteer).deliver if new_record
+    NewEventMailer.invite_volunteer(@event, @volunteer).deliver_later if new_record
   end
 
   def approve
+    @volunteer = current_volunteer
     @volunteer_event.approve!
+    NewEventMailer.approve_volunteer(@event, @volunteer).deliver_later
```

Comment:
think you can just pass in the `current_volunteer` here, because you're not going to reuse it anywhere anyway.